src/test/java/net/snowflake/client/jdbc/ProxyLatestIT.java

```diff
@@ -0,0 +1,346 @@
+    originalTrustStorePath = systemGetProperty(TRUST_STORE_PROPERTY);
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    System.setProperty(
+        TRUST_STORE_PROPERTY, getResourceURL("wiremock" + File.separator + "ca-cert.jks"));
+    startWiremockStandAlone();
+  }
+
+  @After
+  public void tearDown() {
+    stopWiremockStandAlone();
+  }
+
+  @AfterClass
+  public static void tearDownClass() {
+    restoreTrustStorePathProperty();
+  }
+
+  @Test
+  public void testProxyIsUsedWhenSetInProperties() throws SQLException {
+    Properties props = getProperties();
+    addProxyProperties(props);
+    proxyAll(props);
+
+    connectAndVerifySimpleQuery(props);
+    verifyProxyWasUsed();
+  }
+
+  @Test
+  public void testProxyIsUsedWhenSetInJVMParams() throws SQLException {
+    Properties props = getProperties();
+    setJvmProperties(props);
+    proxyAll(props);
+
+    connectAndVerifySimpleQuery(props);
+    verifyProxyWasUsed();
+  }
+
+  @Test
+  public void testProxyNotUsedWhenNonProxyHostsMatchingInProperties() throws SQLException {
+    restoreTrustStorePathProperty();
+    Properties props = getProperties();
+    addProxyProperties(props);
+    props.put("nonProxyHosts", "*");
+    proxyAll(props);
+
+    connectAndVerifySimpleQuery(props);
+    verifyProxyNotUsed();
+  }
+
+  @Test
+  public void testProxyIsUsedWhenNonProxyHostsNotMatchingInProperties() throws SQLException {
+    Properties props = getProperties();
+    addProxyProperties(props);
+    props.put("nonProxyHosts", "notMatchingHost");
+    proxyAll(props);
+
+    connectAndVerifySimpleQuery(props);
+    verifyProxyWasUsed();
+  }
+
+  @Test
+  public void testProxyNotUsedWhenNonProxyHostsMatchingInJVMParams() throws SQLException {
+    restoreTrustStorePathProperty();
+    Properties props = getProperties();
+    setJvmProperties(props);
+    System.setProperty("http.nonProxyHosts", "*");
+    proxyAll(props);
+
+    connectAndVerifySimpleQuery(props);
+    verifyProxyNotUsed();
+  }
+
+  @Test
+  public void testProxyUsedWhenNonProxyHostsNotMatchingInJVMParams() throws SQLException {
+    Properties props = getProperties();
+    setJvmProperties(props);
+    System.setProperty("http.nonProxyHosts", "notMatchingHost");
+    proxyAll(props);
+
+    connectAndVerifySimpleQuery(props);
+    verifyProxyWasUsed();
+  }
+
+  private void startWiremockStandAlone() {
+    // retrying in case of fail in port bindings
+    await()
+        .alias("wait for wiremock responding")
+        .atMost(Duration.ofSeconds(20))
+        .until(
+            () -> {
+              try {
+                httpProxyPort = findFreePort();
+                httpsProxyPort = findFreePort();
+                wiremockStandalone =
+                    new ProcessBuilder(
+                            "java",
+                            "-jar",
+                            getResourceURL(
+                                "wiremock" + File.separator + "/wiremock-standalone-3.7.0.jar"),
```

Comment:
So this basically should work, but for remaining drivers may be more difficult... There is no chance of setting it up as a docker container? Would it work on GH and Jenkins?

Reply:
I tried with docker, but I have no success as we are already running tests in docker images and I couldn't setup running docker in docker correctly. When looking for solution I've found many issues reported for this setup (sometimes related to docker version and os), so it's more difficult option than running a jar. On github runners java is installed by default, for our docker images we could add it.